src/frontend/Navigation/OnboardingStack.tsx

```diff
@@ -16,57 +14,46 @@ import {
   SendJoinRequestScreen,
   SyncOnboardingScreen,
 } from "../screens/Onboarding";
-import CustomHeaderLeft from "../sharedComponents/CustomHeaderLeft";
-import { AppStack } from "./AppStack";
+import { RootStack } from "./AppStack";
 
-const CreateOrJoinStack = createStackNavigator(
-  {
-    CreateOrJoinScreen: CreateOrJoinScreen,
-    JoinProjectQr: JoinProjectQrScreen,
-    SendJoinRequest: SendJoinRequestScreen,
-    Sync: SyncOnboardingScreen,
-  },
-  {
-    initialRouteName: "CreateOrJoinScreen",
-    // TODO iOS: Dynamically set transition mode to modal for modals
-    mode: "card",
-    headerMode: "screen",
-    defaultNavigationOptions: {
-      headerStyle: {
-        height: 60,
-      },
-      // We use a slightly larger back icon, to improve accessibility
-      // TODO iOS: This should probably be a chevron not an arrow
-      headerLeft: props => <CustomHeaderLeft {...props} />,
-      headerTitleStyle: {
-        marginHorizontal: 0,
-      },
-      cardStyle: {
-        backgroundColor: "#ffffff",
-      },
-    },
-  }
-);
-
-const MainStack = createSwitchNavigator(
-  {
-    App: AppStack,
-    CreateOrJoinStack,
-  },
-  {
-    initialRouteName: "App",
-  }
-);
+export type OnboardingStackList = {
+  JoinProjectQrScreen: { isAdmin: boolean };
+  CreateOrJoinScreen: undefined;
+  SendJoinRequestScreen: undefined;
+  SyncOnboardingScreen: { keepExistingObservations: boolean };
+  ProjectInviteModal: { inviteKey: string };
+  JoinRequestModal: { deviceName?: string; key?: string } | undefined;
+};
 
-export const OnboardingStack = createStackNavigator(
-  {
-    Main: MainStack,
-    JoinRequestModal,
-    ProjectInviteModal,
-  },
-  {
-    initialRouteName: "Main",
-    mode: "modal",
-    headerMode: "none",
-  }
+export const OnboardingStack = () => (
```

Comment:
that's good to know, thanks for the explanation!

Reply:
Just for visibility. We actually are NOT going to create a separate stack for onboarding because it does not solve our problem of having the nav state available to the project invite modal. The nav state for children is only available to that child and not the parent or siblings.

 AND following react navigation suggestion, having a nested navigator is superflous unless the UI is different.